Add best practices for managing content (#3004)
theforeman/foreman-documentation: guides/common/modules/ref_best-practices-for-content-views.adoc

[id="best-practices-for-content-views_{context}"]
= Best practices for content views

> * {Team} recommends to use content views that bundle content for {client-os} and additional software such as `Apache-2.4` or `PostgreSQL-16.2`.

Reviewer: ```suggestion
* Create content views that bundle content for {client-os} and additional software such as `Apache-2.4` or `PostgreSQL-16.2`.
```

* Making content management too small-scale results in more manual work.
* If you require daily updated content, use the content view `Default Organization View`, which contains the latest synchronized content from all repositories and is available in the _Library_ lifecycle environment.

> * {Team} recommends using composite content views only if you require more flexibility, for example to update one content view on a weekly basis and another content view on a monthly basis.

Reviewer: ```suggestion
* Restrict composite content views to situations that require greater flexibility, for example, if you update one content view on a weekly basis and another content view on a monthly basis.
```

> * When using composite content views, publish the required content views and then the composite content views afterwards.

Reviewer: ```suggestion
* If you use composite content views, first publish the _required_ content views and then publish the composite content views.
```

> The more content views you bundle, the more effort is needed to change or update content.

Reviewer: ```suggestion
Bundling content views involves a trade-off.
The more content views you bundle, the more effort is needed to change or update content.
```

The more content views you bundle, the more effort is needed to change or update content.

> Make it as simple as possible and as complex as necessary.

Reviewer 2: ```suggestion
```
This doesn't really tell the user how to make it simple or complex. Seems like a filler sentence. Not sure it is needed.

> * Automate creating and publishing composite content views and lifecycle environments using Hammer scripts or Ansible playbooks.

Reviewer: ```suggestion
* Automate creating and publishing composite content views and lifecycle environments by using a Hammer script or the Foreman link:https://theforeman.github.io/foreman-ansible-modules/develop/plugins/content_view_module.html#ansible-collections-theforeman-foreman-content-view-module[manage_content module] in an Ansible playbook.
```

> Use cron jobs or recurring logics for more visibility.

Reviewer 3: systemd timers are also becoming more common.

> Recurring logics require SSH-based access to {Project} and the remote execution plugin.

Reviewer 3: Wouldn't also work with the pull based provider?
```suggestion
Recurring logics requires the remote execution plugin.
```

Reviewer 3: > Maybe drop the sentence altogether because it's a list of best practices, not a procedure?

Yes, probably best to keep that to the procedure.

> * {Team} recommends adding the changes and date to the description of each published content view or composite content view version.

Reviewer 2: ```suggestion
* Add the changes and date to the description of each published content view or composite content view version.
```

> The date in the {ProjectWebUI} shows the latest action, for example promoting content to a different lifecycle environment, which is independent from the latest change to the actual content.

Reviewer 2: ```suggestion
The most recent activity, such as moving content to a new lifecycle environment, is displayed by the date in the {ProjectWebUI}, regardless of the latest changes to the content itself.
```

[id="best-practices-for-content-views_{context}"]
= Best practices for content views

* Create content views that bundle content for {client-os} and additional software such as `Apache-2.4` or `PostgreSQL-16.2`.
Content views that are too small require more effort to maintain.
* If you require daily updated content, use the content view `Default Organization View`, which contains the latest synchronized content from all repositories and is available in the _Library_ lifecycle environment.
* Restrict composite content views to situations that require greater flexibility, for example, if you update one content view on a weekly basis and another content view on a monthly basis.
* If you use composite content views, first publish the content views and then publish the composite content views.

> Bundling content views involves a trade-off.

Reviewer 2: ```suggestion
```
Not sure this sentence is needed.

Author: Dropped.
